llms.txt

```diff
@@ -11289,6 +11290,780 @@ Doc-Content: https://docs.kluster.ai/tutorials/klusterai-api/image-analysis.ipyn
+    "from getpass import getpass\n",
+    "\n",
+    "kluster_api_key = getpass(\"Enter your kluster API key: \")\n",
+    "\n",
+    "# Initialize the client\n",
+    "client = OpenAI(\n",
+    "    base_url=\"https://api.kluster.ai/v1\",  # kluster API endpoint\n",
+    "    api_key=kluster_api_key\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Basic Tool Calling\n",
+    "\n",
+    "Kluster supports tool calling similar to OpenAI's function calling. Let's start with a simple example using a calculator tool. \n",
+    "\n",
+    "Kluster treats the calculator as a first‐class capability you expose to the model: by including its JSON-Schema in the tools array, you tell the LLM, “if the user asks for arithmetic, call this function instead of guessing the answer.” When we send the prompt “What is 1337 × 42?” with `tool_choice=\"auto\"`, the model recognises that the calculator is the best way to satisfy the request and answers not with prose but with a `tool_calls` block that contains the function name and a properly-formatted argument string (\"1337 * 42\")."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\n",
+      "  \"id\": \"chatcmpl-8dd4c1cc-4ecc-4c79-aed6-d408bbad5081\",\n",
+      "  \"choices\": [\n",
+      "    {\n",
+      "      \"finish_reason\": \"tool_calls\",\n",
+      "      \"index\": 0,\n",
+      "      \"logprobs\": null,\n",
+      "      \"message\": {\n",
+      "        \"content\": null,\n",
+      "        \"refusal\": null,\n",
+      "        \"role\": \"assistant\",\n",
+      "        \"audio\": null,\n",
+      "        \"function_call\": null,\n",
+      "        \"tool_calls\": [\n",
+      "          {\n",
+      "            \"id\": \"chatcmpl-tool-fa51f55519ff412bb1798262c3a5a797\",\n",
+      "            \"function\": {\n",
+      "              \"arguments\": \"{\\\"expression\\\": \\\"1337 * 42\\\"}\",\n",
+      "              \"name\": \"calculator\"\n",
+      "            },\n",
+      "            \"type\": \"function\"\n",
+      "          }\n",
+      "        ]\n",
+      "      },\n",
+      "      \"stop_reason\": 128008\n",
+      "    }\n",
+      "  ],\n",
+      "  \"created\": 1746741970,\n",
+      "  \"model\": \"klusterai/Meta-Llama-3.1-8B-Instruct-Turbo\",\n",
+      "  \"object\": \"chat.completion\",\n",
+      "  \"service_tier\": null,\n",
+      "  \"system_fingerprint\": null,\n",
+      "  \"usage\": {\n",
+      "    \"completion_tokens\": 20,\n",
+      "    \"prompt_tokens\": 252,\n",
+      "    \"total_tokens\": 272,\n",
+      "    \"completion_tokens_details\": null,\n",
+      "    \"prompt_tokens_details\": null\n",
+      "  },\n",
+      "  \"prompt_logprobs\": null\n",
+      "}\n"
+     ]
```

Comment:
Ideally we make a statement from the response. How it is structured, and then tell the reader that the next section goes into depth on processing the response of a tool call.